src/mailadm/bot.py

```diff
@@ -0,0 +1,98 @@
+import sys
+
+import deltachat
+from deltachat import account_hookimpl
+from mailadm.db import DB
+from mailadm.commands import add_user, add_token, list_tokens
+import os
+from threading import Event
+
+
+class SetupPlugin:
+    def __init__(self, admingrpid):
+        self.member_added = Event()
+        self.admingrpid = admingrpid
+        self.message_sent = Event()
+
+    @account_hookimpl
+    def ac_member_added(self, chat: deltachat.Chat, contact, actor, message):
+        assert chat.num_contacts() == 2
+        if chat.id == self.admingrpid:
+            self.member_added.set()
+
+    @account_hookimpl
+    def ac_message_delivered(self, message: deltachat.Message):
+        if not message.is_system_message():
+            self.message_sent.set()
```

Comment:
There also seems to be a problem that the `ac_message_delivered` hook isn't triggered in `mailadm setup-bot` after the bot wrote to the newly created group - the message arrives on the other device, but core doesn't notice it.

Reply:
It usually works, just sometimes it fails if there is some db lock left or so. Can't reproduce anymore.

Reply:
Fixed in 360bb5c48d453c6558317ebc8d9895abfe5f0965 :)